src/Skills/Systems/LightningCannonSystem.gd

```diff
@@ -0,0 +1,41 @@
+extends "res://Ecs/Core/System.gd"
+
+var _event_bus
+
+func _init(event_bus):
+	required_component_types = ["HealthComponent"]
+	_event_bus = event_bus
+	_event_bus.connect("lightning_cannon", self, 'activate')
+
+func activate(source_entity, target: Vector2):
+	var target_entity = _get_entity_at(target)
+	if target_entity != null and source_entity != target_entity:
+		_lightning_cannon(source_entity, target_entity, Constants.LIGHTNING_CANNON_DAMAGE, Constants.LIGHTNING_CANNON_RADIUS)
+		_event_bus.emit_signal("end_turn")
+
+func _lightning_cannon(source, target, damage, radius):
+	if damage <= 0 or radius <= 0:
+		return
+	
+	print("cannon: " + str(damage) + ", " + str(radius))
+	_event_bus.emit_signal("damage_entity", target, damage)
+	var target_2 = _get_entity_within_radius(target.position, radius)
+	if target_2 != null and target_2 != target and source != target_2:
+		var distance = target_2.position.distance_to(target.position)
+		var new_damage = int(floor(damage - distance))
```

Comment:
This is an interesting - and good? twist - secondary damage is more effective at close range. May need a `if new_damage > 0` check, though.

Reply:
That is taken care of at the beginning of this function. Recursion!